src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -865,11 +894,72 @@ def process_nw_flow_log_show_namespace(cmd, namespace):
 
 
 def process_nw_topology_namespace(cmd, namespace):
+    from msrestazure.tools import is_valid_resource_id, resource_id, parse_resource_id
+    SubResource = cmd.get_models('SubResource')
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
     location = namespace.location
+    rg = namespace.target_resource_group_name
+    vnet = namespace.target_vnet
+    subnet = namespace.target_subnet
+
+    vnet_id = vnet if is_valid_resource_id(vnet) else None
+    subnet_id = subnet if is_valid_resource_id(subnet) else None
+
+    if rg and not vnet and not subnet:
+        # targeting resource group - OK
+        pass
+    elif subnet:
+        subnet_usage = CLIError('usage error: --subnet ID | --subnet NAME --resource-group NAME --vnet NAME')
+        # targeting subnet - OK
+        if subnet_id and (vnet or rg):
+            raise subnet_usage
+        elif not subnet_id and (not rg or not vnet or vnet_id):
```

Comment:
based on the error message, should we check the subnet , say `not subnet_id and (not rg or not vnet or not subnet or vnet_id)`?

Reply:
I'll test and check.